src/proc/validator.rs

```diff
@@ -430,14 +473,31 @@ impl Validator {
         self.typifier
             .resolve_all(&fun.expressions, &module.types, &resolve_ctx)?;
 
+        if fun.global_usage.len() > module.global_variables.len() {
+            return Err(FunctionError::InvalidGlobalUsageLength);
+        }
+
         for (var_handle, var) in fun.local_variables.iter() {
-            self.validate_local_var(var, fun, &module.types)
+            self.validate_local_var(var, &module.types, &module.constants)
                 .map_err(|error| FunctionError::LocalVariable {
                     handle: var_handle,
                     name: var.name.clone().unwrap_or_default(),
                     error,
                 })?;
         }
+
+        for (index, argument) in fun.arguments.iter().enumerate() {
+            match module.types[argument.ty].inner {
+                crate::TypeInner::Image { .. } | crate::TypeInner::Sampler { .. } => {
```

Comment:
If I may ask, why are Images and Samplers an invalid argument type?

Reply:
Great question! WGSL doesn't support this right now, and in both SPIR-V and GLSL the samplers and images are just globals, so they aren't passed around either. We can come back to this at some point.